src/main/java/org/apache/lucene/analysis/ComboAnalyzer.java

```diff
@@ -168,113 +157,109 @@ public ComboAnalyzer disableDeduplication() {
-
-        // We remember last used TokenStreams because many times Analyzers can provide a reusable TokenStream
-        // Detecting that all sub-TokenStreams are reusable permits to reuse our ComboTokenStream as well.
-        if (tempTokenStreams.get() == null) tempTokenStreams.set(new TokenStream[subAnalyzers.length]); // each time non reusability has been detected
-        if (lastTokenStreams.get() == null) lastTokenStreams.set(new TokenStream[subAnalyzers.length]); // only at first run
-        TokenStream[] tempTokenStreams_local = tempTokenStreams.get();
-        TokenStream[] lastTokenStreams_local = lastTokenStreams.get();
-        ReusableTokenStreamComponents lastComboTokenStream_local = lastComboTokenStream.get();
-        if (lastComboTokenStream_local == null)
-            lastComboTokenStream_local = new ReusableTokenStreamComponents(fieldName, this);
+        @Override
+        public TokenStream getTokenStream() {
+            TokenStream ret = createTokenStreams();
+            return deduplication ? new UniqueTokenFilter(ret): ret;
+        }
 
-        // Get sub-TokenStreams from sub-analyzers
-        for (int i = subAnalyzers.length-1 ; i >= 0 ; --i) {
+        private TokenStream createTokenStreams() {
+            if(subAnalyzers.length == 1){
+                return createTokenStream(subAnalyzers[0], field, reader);
+            }
+            else{
+                ReaderCloneFactory.ReaderCloner<Reader> cloner = ReaderCloneFactory.getCloner(reader);
+                TokenStream[] streams = new TokenStream[subAnalyzers.length];
+                for (int i = 0; i < subAnalyzers.length; i++) {
+                    streams[i] = createTokenStream(subAnalyzers[i], field, cloner.giveAClone());
+                }
+                return new ComboTokenStream(streams);
+            }
+        }
 
-            // Feed the troll
-            Reader reader = readerCloner.giveAClone();
-            tempTokenStreams_local[i] = null;
+        private TokenStream createTokenStream(Analyzer analyzer, String field, Reader reader)  {
             try {
-                tempTokenStreams_local[i] = subAnalyzers[i].tokenStream(fieldName, reader);
-            } catch (IOException ignored) {
-                logger.debug("Ignoring {}th analyzer [{}]. Could not get a TokenStream.", ignored, i, subAnalyzers[i]);
-            }
-            // Use caching if asked or if required in case of duplicated analyzers
-            if (cacheTokenStreams || hasDuplicatedAnalyzers && duplicatedAnalyzers.contains(subAnalyzers[i])) {
-                CachingTokenStream cache = new CachingTokenStream(tempTokenStreams_local[i]);
-                try {
-                    tempTokenStreams_local[i].reset();
-                    cache.fillCache();
-                } catch (IOException ignored) {
-                    logger.debug("Got an error when caching TokenStream from the {}th analyzer [{}]", i, subAnalyzers[i]);
```

Comment:
Rather than making debug outputs, I opted to just throw a runtime exception if something failed. It's quite severe if one looses tokens without getting an error.